include/scalarfield.h

```diff
@@ -0,0 +1,159 @@
+        *   \param MetaDataType The metadata type for this field
+        */
+        ScalarField(const std::string& name, MetaDataType type);
+
+        /*!
+        *   \brief ScalarField constructor with initial vales
+        *   \param name The name used to reference the scalar field
+        *   \param MetaDataType The metadata type for this field
+        *   \param vals Initial values to be copied into the Scalarfield
+        */
+        ScalarField(const std::string& name,
+                    MetaDataType type,
+                    const std::vector<T>& vals);
+
+        /*!
+        *   \brief ScalarField constructor with initial values
+        *   \param name The name used to reference the scalar field
+        *   \param MetaDataType The metadata type for this field
+        *   \param vals Initial values to be place in the Scalarfield
+        *               via move semantics
+        */
+        ScalarField(const std::string& name,
+                    MetaDataType type,
+                    std::vector<T>&& vals);
+
+        /*!
+        *   \brief Default ScalarField copy constructor
+        *   \param scalar_field The scalar field to be copied.
+        */
+        ScalarField(const ScalarField<T>& scalar_field) = default;
+
+        /*!
+        *   \brief Default ScalarField move constructor
+        *   \param scalar_field The scalar field to be moved for construction.
+        */
+        ScalarField(ScalarField<T>&& scalar_field) = default;
+
+        /*!
+        *   \brief Default ScalarField copy assignment operator
+        *   \param scalar_field The scalar field to be copied.
+        */
+        ScalarField<T>& operator=(const ScalarField<T>& scalar_field)
+            = default;
+
+        /*!
+        *   \brief Default ScalarField move assignment operator
+        *   \param scalar_field The scalar field to be moved.
+        */
+        ScalarField<T>& operator=(ScalarField<T>&& scalar_field)
+            = default;
+
+        /*!
+        *   \brief Default MetadataField destructor
+        */
+        virtual ~ScalarField() = default;
+
+        /*!
+        *   \brief Serialize the ScalarField for
+        *          transmission and storage.
+        *   \returns A string of the serialized field
+        */
+        virtual std::string serialize();
+
+        /*!
+        *   \brief Add a value to the field
+        *   \param value A c-ptr to the value to append
+        */
+        void append(const void* value);
+
+        /*!
+        *   \brief Retrieve the number of values in the field
+        *   \returns The number of values
+        */
+        virtual size_t size();
+
+        /*!
+        *   \brief Clear the values in the field
+        */
+        virtual void clear();
+
+        /*!
+        *   \brief Returns a c-ptr to the underlying data.  This
+        *          pointer is not valid if any other operation
+        *          such as append() is performed.
```

Comment:
That is correct.  Calling ``append()`` would make using the returned pointer from ``data()`` unsafe.  As far as I understand, this follows C++ standard container behavior (mainly because we are using standard containers as our underlying data structures).  Because standard containers like ``std::vector`` have to periodically allocate new memory for appended values and move all of the old values to the new memory space, operations like ``append()`` can cause the pointer from ``data()`` to become invalid.  Do you think it is OK that we use this behavior in an internal function?

Reply:
Yeah, sure, it is well documented! I see no problem at all.